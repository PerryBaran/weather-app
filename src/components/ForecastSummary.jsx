import React from "react";
import PropTypes from "prop-types";
import WeatherIcon from "react-icons-weather";

function ForecastSummary({ forecast }) {
  const {
    date,
    description,
    icon,
    temperature: { max: maxTemp },
  } = forecast;

  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-summary" data-testid="forecast-summary">
      <div className="forecast-summary__date">{formattedDate}</div>
      <div className="forecast-summary__icon" data-testid="forecast-icon">
        <WeatherIcon name="owm" iconId={icon} />
      </div>
      <div className="forecast-summary__temperature">{maxTemp}&deg;C</div>
      <div className="forecast-summary__description">{description}</div>
    </div>
  );
}

ForecastSummary.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    description: PropTypes.string,
    icon: PropTypes.string,
    temperature: PropTypes.shape({
      max: PropTypes.number,
    }),
  }).isRequired,
};

export default ForecastSummary;
