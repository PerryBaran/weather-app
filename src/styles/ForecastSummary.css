.forecast-summary {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-items: center;
  align-items: center;
  gap: 10px;
}

.forecast-summary:first-child {
  border-left: none;
}

.forecast-summary__dateOrTime {
  font-size: 1rem;
}

.forecast-summary__icon {
  font-size: min(4rem, 5vw);
}

.forecast-summary__no-data {
  font-size: min(1.5rem, 2vw);
}

.forecast-summary__description {
  font-size: min(1rem, 2.5vw);
}

.forecast-summary__temp {
  font-size: min(1rem, 2.5vw);
}

.forecast-summary__more-details {
  all: unset;
  width: 95%;
  padding: 5px 0;
}

@media screen and (max-width: 780px) {
  .forecast-summary__dateOrTime {
    width: 50px;
  }
}