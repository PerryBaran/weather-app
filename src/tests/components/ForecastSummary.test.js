import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 12345,
    description: "Test description",
    icon: "testIcon",
    temperature: {
      max: 20,
    },
  };

  describe("snapshot", () => {
    it("renders corretly", () => {
      const { asFragment } = render(<ForecastSummary forecast={validProps} />);

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("props", () => {
    it("renders correct values for props", () => {
      const { getByText, getByTestId } = render(<ForecastSummary forecast={validProps} />);

      expect(getByText(validProps.date.toString())).toHaveAttribute(
        "class",
        "forecast-summary__date"
      );
      expect(getByText(validProps.description)).toHaveAttribute(
        "class",
        "forecast-summary__description"
      );
      expect(getByTestId("forecast-icon")).toHaveAttribute(
        "class",
        "forecast-summary__icon"
      );
      expect(getByText(`${validProps.temperature.max}Â°C`)).toHaveAttribute(
        "class",
        "forecast-summary__temperature"
      );
    });
  });
});
