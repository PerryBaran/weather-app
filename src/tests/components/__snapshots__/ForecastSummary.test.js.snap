// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary snapshot renders corretly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-summary__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-summary__icon"
      data-testid="forecast-icon"
    >
      <i
        class="wi wi-day-sunny"
        name="owm"
      />
    </div>
    <div
      class="forecast-summary__temperature"
    >
      20Â°C
    </div>
    <div
      class="forecast-summary__description"
    >
      Test description
    </div>
  </div>
</DocumentFragment>
`;
